### ISSUE TITLE
Microsoft Oauth Username

### ISSUE THREAD

**Issue by ddaviscomsec** [zorn-v/nextcloud-social-login#476]
I have created a custom Oauth for microsoft but when the account is created via furst login the Microsoft GUI is being used as the username. How can I get the email to be used as the username? Can this be included in the standard features?

Thanks!

**Comment by zorn-v** [comment]
No, username in nextcloud considered as user ID.
Customizable of it is not possible intentionally due security issues.
Use "display name claim" option.

**Comment by ddaviscomsec** [comment]
Ok I understand but when MS users authenticate and their account is created the MS GUI is used as their username and this creates an issue in identifying users easily in nextcloud as sometimes only user names are used on certain screens.

**Comment by zorn-v** [comment]
AFAIK display name is preferable in search/display and other things in nextcloud. 
So just figure it out how to use "display name claim"

**Comment by ddaviscomsec** [comment]
I am able to add the user by their display name to a collective as seen below:
![Screenshot 2024-08-19 140253](https://github.com/user-attachments/assets/2a4aa554-b5f0-4d4f-a830-6a54dfe83dc6)


But then when managing the users I can see only the username and not the display name:
![Screenshot 2024-08-19 140311](https://github.com/user-attachments/assets/6f647886-9e9d-4788-98ad-be0fc625038c)



This is why I was asking if there is anyway for me to set the username.

**Comment by zorn-v** [comment]
Yep, probably in other apps it may brings some problems, but it will not changed. 
Thanks for understanding :smile:

**Comment by zorn-v** [comment]
> Is it the claim name ? (like "family_name" or "given_name")

Yep
https://github.com/zorn-v/nextcloud-social-login/blob/249059e320ba7bc9a8b90ea34012388c4f99a293/lib/Provider/CustomOpenIDConnect.php#L34-L38
